packages/wrangler/src/package-manager.ts

```diff
@@ -60,15 +60,20 @@ export async function getPackageManager(cwd: string): Promise<PackageManager> {
     }
   }
 
-  if (hasNpm) {
+  if (process.env.npm_execpath?.includes("npx")) {
```

Comment:
I think we should do similar checks for yarn dlx and pnpm dlx 

Also, wha happens when wrangler has been installed with `npm install -g` or `yarn add global` or `pnpm add -g`?